### PR TITLE
SAAS-3130: Added Instance Media Streams Encrypted Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/connect/instanceMediaStreamsEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                                which is greater than or equal to the desired encryption level ${config.desiredEncryptionLevelString}`,
+                                region, resource);
+                        } else {
+                            helpers.addResult(results, 2,
+                                `Connect instance is using ${currentEncryptionLevelString} for media streams encryption\
+                                which is less than the desired encryption level ${config.desiredEncryptionLevelString}`,
+                                region, resource);
+                        }
+                    } else {
+                        currentEncryptionLevel= 2; //awskms
```
Comment (dipsubha06) [inline plugins/aws/connect/instanceMediaStreamsEncrypted.js:118]: if we are setting the currentEncryptionLevel here this is not being checked after this. Is this okay?

File: plugins/aws/connect/instanceMediaStreamsEncrypted.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    !listInstanceMediaStreamsStorageConfigs.data.StorageConfigs) {
+                    helpers.addResult(results, 3,
+                        `Unable to describe Connect instance media streams storage config: ${helpers.addError(listInstanceMediaStreamsStorageConfigs)}`,
+                        region, resource);
+                    continue;
+                }
+
+                if (!listInstanceMediaStreamsStorageConfigs.data.StorageConfigs.length) {
+                    helpers.addResult(results, 0,
+                        'Connect instance does not have any storage config for media streams',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceMediaStreamsEncrypted.js:79]:
```suggestion
                        'Connect instance does not have any media streams enabled',
```

File: plugins/aws/connect/instanceMediaStreamsEncrypted.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                                `Connect instance is using ${currentEncryptionLevelString} for media streams encryption\
+                                which is less than the desired encryption level ${config.desiredEncryptionLevelString}`,
+                                region, resource);
+                        }
+                    } else {
+                        currentEncryptionLevel= 2; //awskms
+                    }
+                } else {
+                    helpers.addResult(results, 3,
+                        'Unable to find Connect instance media streams Kinesis Video Stream Config',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceMediaStreamsEncrypted.js:122]:
```suggestion
                        'Unable to find Connect instance media streams config',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/connect/instanceMediaStreamsEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'Connect Instance Media Streams Encrypted',
+    category: 'Connect',
+    domain: 'Content Delivery',
+    description: 'Ensure that Amazon Connect instances have encryption enabled for media streams being saved on Kinesis Video Stream.',
+    more_info: 'In Amazon Connect, you can capture customer audio during an interaction with your contact center by sending the audio to a Kinesis video stream.'+
+            'All data put into a Kinesis video stream is encrypted at rest using AWS-managed KMS keys. Use customer-managed keys instead, in order to meet regulatory compliance requirements within your organization.',
```
Comment (AkhtarAmir) [inline plugins/aws/connect/instanceMediaStreamsEncrypted.js:10]:
```suggestion
    more_info: 'In Amazon Connect, you can capture customer audio during an interaction with your contact center by sending the audio to a Kinesis video stream. ' +
            'All data put into a Kinesis video stream is encrypted at rest using AWS-managed KMS keys. Use customer-managed keys instead, in order to meet regulatory compliance requirements within your organization.',
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**